model/histogram/generic.go

```diff
@@ -540,3 +540,86 @@ var exponentialBounds = [][]float64{
 		0.9892280131939752, 0.9919100824251095, 0.9945994234836328, 0.9972960560854698,
 	},
 }
+
+func shrink[IBC InternalBucketCount](originSpans []Span, originBuckets []IBC, originSchema, targetSchema int32, deltaBuckets bool) ([]Span, []IBC) {
```

Comment:
Would the method approach I suggested in the other comment help you? As said, I would like to avoid exposing a generic function that looks quite low level to me.